src/main/java/io/vertx/ext/httpservicefactory/HttpServiceFactory.java

```diff
@@ -0,0 +1,388 @@
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+/**
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class HttpServiceFactory extends ServiceVerticleFactory {
+
+  public static final String CACHE_DIR_PROPERTY = "vertx.httpServiceFactory.cacheDir";
+  public static final String HTTP_CLIENT_OPTIONS_PROPERTY = "vertx.httpServiceFactory.httpClientOptions";
+  public static final String HTTPS_CLIENT_OPTIONS_PROPERTY = "vertx.httpServiceFactory.httpsClientOptions";
+  public static final String AUTH_USERNAME_PROPERTY = "vertx.httpServiceFactory.authUsername";
+  public static final String AUTH_PASSWORD_PROPERTY = "vertx.httpServiceFactory.authPassword";
+  public static final String KEYSERVER_URI_TEMPLATE = "vertx.httpServiceFactory.keyserverURITemplate";
+  public static final String VALIDATION_POLICY = "vertx.httpServiceFactory.validationPolicy";
+
+  private static final String FILE_SEP = System.getProperty("file.separator");
+  private static final String FILE_CACHE_DIR = ".vertx" + FILE_SEP + "vertx-http-service-factory";
+
+  private Vertx vertx;
+  private File cacheDir;
+  private String username;
+  private String password;
+  private String keyserverURITemplate;
+  private ValidationPolicy validationPolicy;
+  private HttpClientOptions options;
+
+  @Override
+  public void init(Vertx vertx) {
+    String path = System.getProperty(CACHE_DIR_PROPERTY, FILE_CACHE_DIR);
+    cacheDir = new File(path);
+    cacheDir.mkdirs();
+    options = configOptions();
+    validationPolicy = ValidationPolicy.valueOf(System.getProperty(VALIDATION_POLICY).toUpperCase());
+    username = System.getProperty(AUTH_USERNAME_PROPERTY);
+    password = System.getProperty(AUTH_PASSWORD_PROPERTY);
+    keyserverURITemplate = System.getProperty(KEYSERVER_URI_TEMPLATE, "http://pool.sks-keyservers.net:11371/pks/lookup?op=get&options=mr&search=0x%016X");
+    this.vertx = vertx;
+  }
+
+  protected HttpClientOptions createHttpClientOptions(String scheme) {
+    if ("https".equals(scheme)) {
+      String optionsJson = System.getProperty(HTTPS_CLIENT_OPTIONS_PROPERTY);
+      HttpClientOptions options;
+      if (optionsJson != null) {
+        options = new HttpClientOptions(new JsonObject(optionsJson));
+      } else {
+        options = createHttpClientOptions("http").setTrustAll(true);
+      }
+      options.setSsl(true);
+      return options;
+    } else {
+      String optionsJson = System.getProperty(HTTP_CLIENT_OPTIONS_PROPERTY);
+      return optionsJson != null ? new HttpClientOptions(new JsonObject(optionsJson)) : new HttpClientOptions();
+    }
+  }
+
+  protected HttpClientOptions configOptions() {
+    return createHttpClientOptions(prefix());
+  }
+
+  @Override
+  public String prefix() {
+    return "http";
+  }
+
+  @Override
+  public void resolve(String identifier, DeploymentOptions deploymentOptions, ClassLoader classLoader, Future<String> resolution) {
+
+    int pos = identifier.lastIndexOf("::");
```

Comment:
The docs don't seem to explain anywhere what format is expected... Something containing "::" ??